content/posts/2021-06-25-Treasury-Expenditure-Bug.md

```diff
@@ -0,0 +1,167 @@
+
+  - A security and safety mechanism, so that in case of implementation bugs or compromise of treasury keys the treasury account wouldn't be depleted too quickly, before the issue could be addressed.
+  - A generic "fiscal" policy over the treasury's funds, so that it would not be amenable to being overly spent by short term interests.
+
+This restriction would be enforced via consensus rules in the software, and thus be unchangeable without some other voting process taking place and stakeholders weighing in on the matter.
+
+It's important to note here that no proposal is fully detailed to the level required by actual software code and even this seemingly simple restriction has ambiguity: Decred's blockchain has no concept of "month" and standard calendar months are not easily transposable to a window of blockchain blocks of fixed size. Nevertheless, developers and stakeholders assumed this restriction could be implemented after being properly specified in terms of blockchain primitives.
+
+However, during development of the treasury feature, at this point in time at around October/2020, two objections were raised against this specific form of expenditure policy:
+
+  - The treasury was already spending more than it was receiving via the block reward, due to the USD/DCR exchange rate. Any further decrease would already make it impossible to pay for all monthly contractor invoices.
+  - This policy is not stable over a long time frame, since the block reward is being constantly reduced. Following it strictly would mean the treasury would be unable to spend most of its funds, even if it had accumulated a large pool, since the per-month income would be small.
+
+Therefore, a change in the expenditure policy was proposed and discussed among the developers. The general wording for this policy is noted in the [source code](https://github.com/decred/dcrd/blob/afff2fdbcd4c57ade4f0d13e78ad2d3efaebcdec/blockchain/treasury.go#L651-L655):
+
+> The sum of tspends inside an expenditure window cannot exceed the average of the tspends in the previous N windows in addition to an X% increase.
+
+A `window` here is a specific number of blocks, as defined by some constants in the [chaincfg](https://github.com/decred/dcrd/blob/afff2fdbcd4c57ade4f0d13e78ad2d3efaebcdec/chaincfg/mainnetparams.go#L395-L411) package. For mainnet, this cashes out as roughly:
+
+> The sum of tspends within a 6912 window (aprox. 24 days) cannot exceed the average of the preceding 41472 blocks (aprox. 144 days or 4.8 30-day months) in addition to a 50% increase.
+
+This wording intended to preserve the original safety and policy characteristics as defined in the Politeia proposal, by capping rate at which the expenditures could grow, but still allowing enough flexibility that it could handle cases where the exchange rate dropped significantly or that the treasury had enough accumulated funds that stakeholders felt comfortable spending a larger fraction of it.
```

Comment:
```suggestion
This wording intended to preserve the original safety and policy characteristics as defined in the Politeia proposal, by capping the rate at which expenditures could grow, but still allowing enough flexibility that it could handle cases where the exchange rate dropped significantly or that the treasury had enough accumulated funds that stakeholders felt comfortable spending a larger fraction of it.
```